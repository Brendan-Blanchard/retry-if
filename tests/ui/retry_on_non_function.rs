use retry_if::{retry, ExponentialBackoffConfig};
use std::time::Duration;

const BACKOFF_CONFIG: ExponentialBackoffConfig = ExponentialBackoffConfig {
    max_retries: 5,
    t_wait: Duration::from_secs(1),
    backoff: 2.0,
    t_wait_max: None,
    backoff_max: None,
};

fn should_retry(_: ()) -> bool {
    true
}

#[retry(BACKOFF_CONFIG, should_retry)]
pub trait NotAFunction {
    fn some_function(&mut self);
}

fn main() {}
