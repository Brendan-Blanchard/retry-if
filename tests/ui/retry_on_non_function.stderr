error: custom attribute panicked
  --> tests/ui/retry_on_non_function.rs:16:1
   |
16 | #[retry(BACKOFF_CONFIG, retry_if)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: failed to parse item under #[retry(...)] as function: Error("expected `fn`")
